pkg/core/logs/matcher.go

```diff
@@ -0,0 +1,141 @@
+package logs
+
+import (
+	"context"
+	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
+	"github.com/Kong/kuma/pkg/core"
+	mesh_core "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
+	"github.com/Kong/kuma/pkg/core/resources/manager"
+	"github.com/Kong/kuma/pkg/core/resources/store"
+	"github.com/pkg/errors"
+)
+
+var logger = core.Log.WithName("logs")
+
+// Current limitations:
+// 1) Right now we only generate and place access logs for outbound listeners
+// 2) We match all tags in source section of TrafficLog but only service tag on destination
+// 3) Let's assume we've got following dataplanes:
+//    Dataplane 1 with services: kong and kong-admin
+//    Dataplane 2 with services: backend
+//    If we define rule kong->backend, it is also applied for kong-admin because there is no way to differentiate
+//    traffic from services that are using one dataplane.
+type TrafficLogsMatcher struct {
+	ResourceManager manager.ResourceManager
+}
+
+type MatchedLogs struct {
+	Outbounds map[string][]*mesh_proto.LoggingBackend
```

Comment:
as far as I understand, `[]*mesh_proto.LoggingBackend` has no consistent order.

It might cause unnecessary Envoy configuration updates and flaky unit tests.